python/cudf/cudf/core/frame.py

```diff
@@ -97,6 +97,133 @@ def _as_column(self):
 
         return self._data[None].copy(deep=False)
 
+    def _scatter(self, key, value):
+        result = self._from_table(libcudfxx.copying.scatter(value, key, self))
+
+        result._copy_categories(self)
+        return result
+
+    def _empty_like(self):
+        result = self._from_table(libcudfxx.copying.table_empty_like(self))
+
+        result._copy_categories(self)
+        return result
+
+    def _slice(self, arg):
+        """
+       _slice : slice the frame as per the arg
+
+       Parameters
+       ----------
+       arg : should always be of type slice and doesn't handle step
```

Comment:
Should we throw in the function if the stepsize is not `None` or `1` for now?

Reply:
Yes, I will add that